src/flake8_simplify/plugins/ast_bool_op.rs

```diff
@@ -143,6 +145,29 @@ pub fn duplicate_isinstance_call(checker: &mut Checker, expr: &Expr) {
     }
 }
 
+define_violation!(
+    pub struct AAndNotA(String);
+);
+
+impl Violation for AAndNotA {
```

Comment:
I think for now I'd prefer that these are still defined in `registry.rs`, even if they're defined as standalone structs that implement `Violation`. Wdyt?

Reply:
This would certainly simplify the conversion of `registry.rs` since then the conversion script wouldn't have to deal with editing multiple files / module visibility and adding imports. So yes I think for the initial conversion that makes sense ... nice idea, I didn't consider that.

We can then gradually move the structs & Violation impls to the other modules ... do you agree?

Reply:
I just realized that there is a slight drawback ... we have to make all the struct fields public because otherwise the other modules cannot construct the structs.

And we still have to update every `Check::new` call to construct these new structs (and import them) and call `.into()`.

Reply:
Could we instead introduce a `registry.rs` in each plugin folder (like `flake8_simplify/registry.rs`), and define the checks for each plugin within that crate? Maybe make the fields `pub(crate)`? Or does that have the same issues?

Reply:
I would prefer to deal with the moving of the new structs/impls to other modules in a followup PR ... this PR is already huge.